main/utils/category.py

```diff
@@ -0,0 +1,49 @@
+from typing import Annotated
+
+from fastapi import Depends
+
+from main.commons.exceptions import (
+    BadRequest,
+    ErrorCode,
+    ErrorMessage,
+    Forbidden,
+    NotFound,
+)
+from main.engines.categories import (
+    get_category_by_id,
+    get_category_by_name,
+)
+from main.models.category import CategoryModel
+from main.utils.auth import require_authentication
+
+
+async def get_category_or_404(id: int) -> CategoryModel:
+    category = await get_category_by_id(id)
+
+    if not category:
+        raise NotFound()
+
+    return category
+
+
+async def require_category_creator(
+    category_id: int,
+    id: Annotated[int, Depends(require_authentication)],
```

Comment:
The variable name `id` is too generic, it's also a built-in function. Let's rename it to `user_id`

Reply:
Fixed